### DISCUSSION TITLE
PhantomCamera2D "Framed" Function doesnt work

### DISCUSSION THREAD

**Discussion by DeliMysticShelter** [ramokz/phantom-camera#400] [Question / Help Wanted]
Hello,
i didn't really make anything special, i've got a phantom camera, which is attached to my rigidbody player, the phantom Camera is told to frame a 2d Maker which is a child of my player Rigidbody
It just ignores the Framing, it works exactly like i would use the "Glued" Function

Is this a known Bug?

**Comment by ramokz** [comment]
This is the first time I've heard about an issue like this. Can you share any more details about your setup?

**Reply by DeliMysticShelter** [reply]
Ofcourse! Thank you for answering! 

My Camera2D is Child of the CharacterBody2D.
PhantomCameraHost is Child of the Camera2D.

A PhantomCamera is Child of the CharacterBody2D, in FrameMode is it supposed to follow a Marker2D, which is a Child of the CharacterBody2D.

The Camera2D has a Script for stuff like Screenshake, setting it's offsets

My Bigger Setup is to have various Phantom Cameras in my Level for different Zones, with the possibility to switch Priority to the PhantomCamera, which is Child of the CharacterBody2D.

**Reply by ramokz** [reply]
Can't immediately tell what the issue might be, but would recommend try to enable the `Top Level` property on the `PCam2D` and `Camera2D` nodes.
![image](https://github.com/user-attachments/assets/0f175d91-7e3d-4141-a532-4183ca4dcb9a)

Mainly because the tree structure that you're describing suggests that it might be due to the `PCam2D` and `Camera2D` are child nodes of the `CharacerBody2D`. In general, they should be top-level nodes so they don't get moved around by their parent nodes, which can cause unwanted behavior. Enabling the `Top Level` property on the `PCam2D` and `Camera2D` _should_ circumvent that.

**Reply by DeliMysticShelter** [reply]
This did work, thank you so much. And sorry for posting it as a Problem when my Scene was'nt set up properly. 
With the Top Level Property checked, it does work as intended and helped me to not completely rearrange my Camera Setup.
Only thing i get now is the Problem of Framed Follow staying in the Offset Position when the Camera changes, but that's a known Bug and i already found another Discussion about it.

Thank you!

**Reply by ramokz** [reply]
No problem!
I had thought about applying this automatically, but there are a few use cases issues with doing that.
But sleeping on it some more, I think there's a way to avoid those issues and make it work out-of-the-box — so thanks for posting the question!
